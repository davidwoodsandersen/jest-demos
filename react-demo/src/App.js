import React, { useState } from 'react';
import './App.scss';
import TodoList from './components/TodoList/TodoList';
import Form from './components/Form/Form';

function App(props) {
  const [todos, setTodos] = useState([]);
  const addTodo = (input) => {
    todos.push(input.value);
    setTodos(todos.slice(0));
  };
  
  return (
    <div className="App">
      <h1>React Demo</h1>
      <TodoList todos={todos} />
      <Form addTodo={addTodo} />
    </div>
  );
}

export default App;
