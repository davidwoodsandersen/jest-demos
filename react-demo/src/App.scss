.App {
  display: block;
  margin: 0 auto;
  max-width: 900px;
  padding: 30px;
  background-color: #f5f5f5;
  border-radius: 6px;

  h1 {
    margin: 0 0 20px;
  }
}
