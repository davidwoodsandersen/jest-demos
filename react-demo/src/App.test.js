import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const instance = renderer.create(<App />).toJSON();
  expect(instance).toMatchSnapshot();
});
