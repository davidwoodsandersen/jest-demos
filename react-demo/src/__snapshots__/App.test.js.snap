// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    React Demo
  </h1>
  <div
    className="TodoList"
  >
    <h3>
      Todo List
    </h3>
  </div>
  <form
    className="Form"
    onSubmit={[Function]}
  >
    <input
      type="text"
    />
    <button
      className="Button"
    >
      Add Todo
    </button>
  </form>
</div>
`;
